lib/rubocop/cop/rails/enum_uniqueness.rb

```diff
@@ -32,15 +48,17 @@ class EnumUniqueness < Base
         PATTERN
 
         def on_send(node)
-          enum?(node) do |pairs|
+          enum?(node) do |key, args|
+            consecutive_duplicates(args.values).each do |item|
+              add_offense(item, message: message(key, item))
+            end
+          end
+
+          enum_with_old_syntax?(node) do |pairs|
             pairs.each do |pair|
               enum_values(pair) do |key, args|
-                items = args.values
-
-                next unless duplicates?(items)
-
-                consecutive_duplicates(items).each do |item|
-                  add_offense(item, message: format(MSG, value: item.source, enum: enum_name(key)))
+                consecutive_duplicates(args.values).each do |item|
+                  add_offense(item, message: message(key, item))
```

Comment:
refactor to suppress Metrics/AbcSize and Metrics/MethodLength offences.

<details><summary>Details</summary>

```
$ bundle exec rubocop lib/rubocop/cop/rails/enum_uniqueness.rb
Inspecting 1 file
C

Offenses:

lib/rubocop/cop/rails/enum_uniqueness.rb:50:9: C: Metrics/AbcSize: Assignment Branch Condition size for on_send is too high. [<9, 19, 4> 21.4/17]
        def on_send(node) ...
        ^^^^^^^^^^^^^^^^^
lib/rubocop/cop/rails/enum_uniqueness.rb:50:9: C: Metrics/MethodLength: Method has too many lines. [16/14]
        def on_send(node) ...
        ^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected


$ git diff lib/rubocop/cop/rails/enum_uniqueness.rb 
diff --git a/lib/rubocop/cop/rails/enum_uniqueness.rb b/lib/rubocop/cop/rails/enum_uniqueness.rb
index 2885315a..4b1e4230 100644
--- a/lib/rubocop/cop/rails/enum_uniqueness.rb
+++ b/lib/rubocop/cop/rails/enum_uniqueness.rb
@@ -56,9 +56,13 @@ module RuboCop
 
           enum_with_old_syntax?(node) do |pairs|
             pairs.each do |pair|
+              items = args.values
+
+              next unless duplicates?(items)
+
               enum_values(pair) do |key, args|
                 consecutive_duplicates(args.values).each do |item|
-                  add_offense(item, message: message(key, item))
+                  add_offense(item, message: format(MSG, value: item.source, enum: enum_name(key)))
                 end
               end
             end
```

</details>